imageio/__main__.py

```diff
@@ -0,0 +1,137 @@
+""" 
+Console scripts and associated helper methods for imageio.
+"""
+from __future__ import print_function
+
+import argparse
+import os
+from os import path as op
+import shutil
+
+
+from . import plugins
+from .core import util
+
+# A list of plugins that require binaries from the imageio-binaries
+# repository. These plugins must implement the `download` method.
+PLUGINS_WITH_BINARIES = ["avbin", "ffmpeg", "freeimage"]
+
+
+def download_bin(plugin_names=["all"], package_dir=False):
+    """ Download binary dependencies of plugins
+    
+    This is a convenience method for downloading the binaries
+    (e.g. `ffmpeg.win32.exe` for Windows) from the imageio-binaries
+    repository.
+    
+    Parameters
+    ----------
+    plugin_names: list
+        A list of imageio plugin names. If it contains "all", all
+        binary dependencies are downloaded.
+    package_dir: bool
+        If set to `True`, the binaries will be downloaded to the
+        `resources` directory of the imageio package instead of
+        to the users application data directory. Note that this
+        might require administrative rights if imageio is installed
+        in a system directory.
+    """
+    if plugin_names.count("all"):
+        # Use all plugins
+        plugin_names = PLUGINS_WITH_BINARIES
+    
+    plugin_names.sort()
+    print("Ascertaining binaries for: {}.".format(", ".join(plugin_names)))
+    
+    if package_dir:
+        # Download the binaries to the `resources` directory
+        # of imageio. If imageio comes as an .egg, then a cache
+        # directory will be created by pkg_resources (requires setuptools).
+        # see `imageio.core.util.resource_dirs`
+        # and `imageio.core.utilresource_package_dir`
+        directory = util.resource_package_dir()
+    else:
+        directory = None
+    
+    for plg in plugin_names:
+        msg = "Plugin {} is not registered for binary download!".format(plg)
+        assert plg in PLUGINS_WITH_BINARIES, msg
```

Comment:
Can you make this a test + raise, assert statements are removed from optimized bytecode.